src/generator/writers/writeStdlib.ts

```diff
@@ -1223,4 +1223,20 @@ export function writeStdlib(ctx: WriterContext) {
             `);
         });
     });
+
+    ctx.fun(`__tact_log2`, () => {
+        ctx.signature(`int __tact_log2(int num)`);
+        ctx.context('stdlib');
+        ctx.asm(`asm "UBITSIZE DEC"`);
+    });
+
+    ctx.fun(`__tact_log`, () => {
+        ctx.signature(`int __tact_log(int num, int base)`);
+        ctx.context('stdlib');
+        ctx.body(() => {
+            ctx.write(`
+                return __tact_log2(num) / __tact_log2(base);
```

Comment:
This formula does not work for integer logarithms with an arbitrary base. We basically need `floor(log(num, base))`, which, for example, for `log3(2^32 - 1)` gives us 20 as the result, not 32, which this formula provides.